[ILM] Show forcemerge in hot when rollover is searchable snapshot is enabled

jloleysens opened the pull request:
## Summary

* Implement stakeholder feedback to show forcemerge in hot even when Searchable Snapshot (SS) is on
* Fix bug with validation on policy that does not have rollover and does not block saving when rollover is enabled with no values

<details>

<summary>Gif of SS</summary>

![SS-field](https://user-images.githubusercontent.com/8155004/101505978-32baec80-3975-11eb-8a5c-cc3afabc69db.gif)

<details>

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92771/)
* Commit: a0c382ac258985e6ee3e6b02bfdc000276e204de

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6ff71992a3db78def1b3b9345605714ac5bc110b) | [after](https://github.com/elastic/kibana/commit/a0c382ac258985e6ee3e6b02bfdc000276e204de) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 242.8KB | 243.1KB | +263.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/6ff71992a3db78def1b3b9345605714ac5bc110b) | [after](https://github.com/elastic/kibana/commit/a0c382ac258985e6ee3e6b02bfdc000276e204de) | diff |
| --- | --- | --- | --- |
| `default` | 46960 | 47720 | +760 |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92771/","number":"92771","commit":"a0c382ac258985e6ee3e6b02bfdc000276e204de"}]}
PIPELINE-->

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/validations.ts:
```diff
@@ -56,33 +56,29 @@ export const ROLLOVER_EMPTY_VALIDATION = 'ROLLOVER_EMPTY_VALIDATION';
-        message: i18nTexts.editPolicy.errors.maximumSizeRequiredMessage,
-      },
-    ]);
+    return path === ROLLOVER_FORM_PATHS.maxAge
```
nit: might be a little easier to read using `if...else` instead of a nested ternary.

cjcenizal replied:
A switch/case could also work.

jloleysens replied:
@cjcenizal in this case the switch statement would look something like:

```ts
    switch (true) {
      case 1 === 1:
        return 'ok';
      case 2 === 2:
        return 'ok then';
      default:
        return 'not ok';
    }
```

Just curious to hear thoughts on that pattern where we have `true` as the deciding condition and predicate expressions per case. IMO it might be less easy to grok for the majority of readers because it is less common - WDYT?

jloleysens replied:
My take is it does look pretty neat though!

cjcenizal replied:
@jloleysens I was thinking something more like this unless I'm missing something:

```js
switch (path) {
  case ROLLOVER_FORM_PATHS.maxAge:
    return {
      code: ROLLOVER_EMPTY_VALIDATION,
      message: i18nTexts.editPolicy.errors.maximumAgeRequiredMessage,
    };

  case ROLLOVER_FORM_PATHS.maxDocs:
    return {
      code: ROLLOVER_EMPTY_VALIDATION,
      message: i18nTexts.editPolicy.errors.maximumDocumentsRequiredMessage,
    };

  default:
    return {
      code: ROLLOVER_EMPTY_VALIDATION,
      message: i18nTexts.editPolicy.errors.maximumSizeRequiredMessage,
    };
}
```

jloleysens replied:
Ah OK, I had a different idea. Yours is simpler!

alisonelizabeth reviewed (approved):
Code LGTM. Tested locally. Great catch on the validation bug! Is there a test that we could add that might have caught this?

jloleysens commented:
@alisonelizabeth thanks for the review! I've addressed your feedback in my latest commit.

> Is there a test that we could add that might have caught this?

Yes, but I don't think adding a test at the level of this plugin is the correct place. This plugin does specify a test for the validation on rollover showing under expected conditions:

`should show errors when trying to save with no max size, no max age and no max docs`

That should theoretically have covered this, but because of how the form lib works with validation on submit, this was not caught. So I pivoted to using the form lib in a way that it does work, we should probably add a test case for this in the form lib - I'll open an issue for it.

jloleysens commented:
@alisonelizabeth issue here: https://github.com/elastic/kibana/issues/85383

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93020/)
* Commit: 1f02f172bd1b7df45f8e60992c3fe38f143fcf97

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6ff71992a3db78def1b3b9345605714ac5bc110b) | [after](https://github.com/elastic/kibana/commit/1f02f172bd1b7df45f8e60992c3fe38f143fcf97) | diff |
| --- | --- | --- | --- |
| `indexLifecycleManagement` | 242.8KB | 243.1KB | +293.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/6ff71992a3db78def1b3b9345605714ac5bc110b) | [after](https://github.com/elastic/kibana/commit/1f02f172bd1b7df45f8e60992c3fe38f143fcf97) | diff |
| --- | --- | --- | --- |
| `default` | 46960 | 47720 | +760 |

### History
* :green_heart: [Build #92771](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92771/) succeeded a0c382ac258985e6ee3e6b02bfdc000276e204de

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/92771/","number":"92771","commit":"a0c382ac258985e6ee3e6b02bfdc000276e204de"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93020/","number":"93020","commit":"1f02f172bd1b7df45f8e60992c3fe38f143fcf97"}]}
PIPELINE-->
